### PR TITLE
fix: stub performance.getEntriesByName

### PR DESCRIPTION
**Pull request by liz3**
This stub implements `performance.getEntriesByName` to return a empty array. I'm not famiilar enough yet with the codebase to entirely implement the api but this might be a short term solution.

Should fix(i didn't verify this): https://github.com/oven-sh/bun/issues/6537

### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
I tested that the function does return a empty array.

### CODE REVIEW

File: src/bun.js/bindings/ZigGlobalObject.cpp [hunk 1]
```diff
@@ -2213,9 +2213,17 @@ static inline EncodedJSValue functionPerformanceNowBody(JSGlobalObject* globalOb
     return JSValue::encode(jsDoubleNumber(result));
 }
 
+static inline EncodedJSValue functionPerformanceGetEntriesByNameBody(JSGlobalObject* globalObject) {
+    auto& vm = globalObject->vm();
+    auto* global = reinterpret_cast<GlobalObject*>(globalObject);
+    auto array = JSC::JSArray::create(vm, global->arrayStructureForIndexingTypeDuringAllocation(JSC::ArrayWithContiguous), 0);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/ZigGlobalObject.cpp]:
```suggestion
    auto* array = JSC::constructEmptyArray(globalObject, nullptr);
```

File: src/bun.js/bindings/ZigGlobalObject.cpp [hunk 2]
```diff
@@ -2276,6 +2284,10 @@ class JSPerformanceObject final : public JSC::JSNonFinalObject {
         this->putDirect(vm, JSC::Identifier::fromString(vm, "mark"_s), noopNotImplemented, JSC::PropertyAttribute::Function | 0);
         this->putDirect(vm, JSC::Identifier::fromString(vm, "markResourceTiming"_s), noopNotImplemented, JSC::PropertyAttribute::Function | 0);
         this->putDirect(vm, JSC::Identifier::fromString(vm, "measure"_s), noopNotImplemented, JSC::PropertyAttribute::Function | 0);
+        // this is a stub impl for getEntriesByName(https://github.com/oven-sh/bun/issues/6537)
+        JSFunction* getEntriesByName = JSFunction::create(
+            vm, globalObject(), 0, String("getEntriesByName"_s), functionPerformanceGetEntriesByName, ImplementationVisibility::Public, NoIntrinsic, functionPerformanceGetEntriesByName, nullptr);
+        this->putDirect(vm, JSC::Identifier::fromString(vm, "getEntriesByName"_s), getEntriesByName, JSC::PropertyAttribute::Function | 0);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/ZigGlobalObject.cpp]:
```suggestion
        this->putDirect(vm, JSC::Identifier::fromString(vm, "getEntriesByName"_s), getEntriesByName, 0);
```

**Comment by orpos** [comment]
when i tried to simulate the behaviour of it returning an empty array it worked and started the server
```
$ next dev
  ▲ Next.js 13.5.5
  - Local:        http://localhost:3000

 ✓ Ready in NaNms
```
 it will show this when starting but for now it seems okay to return an empty array

File: src/bun.js/bindings/ZigGlobalObject.cpp (hunk 1 above)
Comment (liz3) [reply src/bun.js/bindings/ZigGlobalObject.cpp]: done!

**Comment by Electroid** [comment]
Thanks for fixing this. We'll need to actually implement this, but at least the stub will make Next.js run again.

**Comment by patelharsh9797** [comment]
How to make it work?
I'm still getting the same performance.getEntriesByName error. Do i need to do something with bun or t should be with nextjs?

Also when i run command bun dev it only show one line with next dev and do nothing. it only show one line of that and that's it. [https://ibb.co/YjWvkcz](url)

What should i do?

**Comment by Electroid** [comment]
> How to make it work? I'm still getting the same performance.getEntriesByName error. Do i need to do something with bun or t should be with nextjs?
> 
> Also when i run command bun dev it only show one line with next dev and do nothing. it only show one line of that and that's it. [https://ibb.co/YjWvkcz](url)
> 
> What should i do?

Try bun with:

```
bun upgrade --canary
```

We haven't released this yet.

**Comment by Gautamvaishnav-git** [comment]
> > How to make it work? I'm still getting the same performance.getEntriesByName error. Do i need to do something with bun or t should be with nextjs?
> > Also when i run command bun dev it only show one line with next dev and do nothing. it only show one line of that and that's it. [https://ibb.co/YjWvkcz](url)
> > What should i do?
> 
> Try bun with:
> 
> ```
> bun upgrade --canary
> ```
> 
> We haven't released this yet.

It works

**Pull request merged**